psakefile.ps1

```diff
@@ -16,33 +16,43 @@ task Clean {
   }
 }
 
-task VendorEditorServices {
-  $githubref = $config.editorComponents.editorServices.githubref
-  if ($config.editorComponents.editorServices.githubuser) {
-    $githubuser = $config.editorComponents.editorServices.githubuser
-  }
-  else {
-    $githubuser = 'lingua-pupuli'
-  }
-  if ($config.editorComponents.editorServices.githubrepo) {
-    $githubrepo = $config.editorComponents.editorServices.githubrepo
-  }
-  else {
-    $githubrepo = 'puppet-editor-services'
+# "editorServices": {
+#   "release": "0.26.0"
+# }
+# "editorServices": {
+#   "release": "0.26.0"
+#   "githubrepo": "puppet-editor-services",
+#   "githubuser": "glennsarti"
+# }
+  # "editorServices": {
+#   "githubrepo": "puppet-editor-services",
+#   "githubref": "glennsarti:spike-rearch-langserver"
+# },
+task VendorEditorServices -precondition { !(Test-Path (Join-Path $PSScriptRoot 'vendor/languageserver')) }{
+  $githubrepo = $config.editorComponents.editorServices.githubrepo ?? 'puppet-editor-services'
+  $githubuser = $config.editorComponents.editorServices.githubuser ?? 'puppetlabs'
+
+  if($config.editorComponents.editorServices.release){
+    $releasenumber = $config.editorComponents.editorServices.release
+    $uri = "https://github.com/${githubuser}/${githubrepo}/releases/download/${releasenumber}/puppet_editor_services_${releasenumber}.zip";
+  }else{
+    $githubref = $config.editorComponents.editorServices.githubref;
+    if($githubref -notcontains ':'){
+      throw "Invalid githubref. Must be in user:branch format like glennsarti:spike-rearch-langserver"
+    }
+    $githubuser = $githubref.split(":")[0]
+    $githubbranch = $githubref.split(":")[1]
+    $uri = "https://github.com/${githubuser}/${githubrepo}/archive/${githubbranch}.zip"
   }
 
   if ($config.editorComponents.editorServices.directory) {
     Copy-Item -Path $config.editorComponents.editorServices.directory -Destination $languageServerPath -Recurse -Force
   }elseif ($config.editorComponents.editorServices.release) {
```

Comment:
nit: Inconsistent bracket style.  I know this isn't part of this PR and I'd be happy to merge it even if not fixed.

Reply:
Fixed.